SimPEG/Maps.py

```diff
@@ -387,6 +387,171 @@ def nP(self):
+        #         len(arg) == 2 and
+        #         isinstance(arg[0], 'Mesh') and
+        #         # TODO: this should be extended to a slice.
+        #         (arg[1].shape[0]==arg[0].nC)
+        #     ), (
+        #         "Each wire needs to be a tuple: (Mesh, Active). "
+        #         "You provided: {}".format(arg)
+        #     )
+
+        self.meshGlobal = args[0][0]
+        self.actvGlobal = args[0][1]
+
+        self.meshLocal = args[1][0]
+        self.actvLocal = args[1][1]
+
+    @property
+    def tree(self):
+        """
+            Create cKDTree structure for given global mesh
+        """
+        if getattr(self, '_tree', None) is None:
+            self._tree = cKDTree(np.c_[self.meshGlobal.gridCC[self.actvGlobal, 0],
+                                       self.meshGlobal.gridCC[self.actvGlobal, 1],
+                                       self.meshGlobal.gridCC[self.actvGlobal, 2]])
+
+        return self._tree
+
+    @property
+    def P(self):
+        """
+            Set the projection matrix with partial volumes
+        """
+        if getattr(self, '_P', None) is None:
+            indx = self.getTreeIndex(self.tree, self.meshLocal, self.actvLocal)
+
+            # Get the node coordinates (bottom-SW) and (top-NE) of cells
+            glob_bsw, glob_tne = self.getNodeExtent(self.meshGlobal)
+
+            loca_bsw, loca_tne = self.getNodeExtent(self.meshLocal)
+
+            # Calculate interesected volume
+            V = []
+            indx = np.r_[indx]
+            for ii in range(indx.shape[1]):
+
+                # Grab corners for ith nearest cell
+                nbsw = glob_bsw[indx[:, ii], :]
+                ntne = glob_tne[indx[:, ii], :]
+
+                V += [(np.max([np.min([ntne[:, 0], loca_tne[:, 0]], axis=0)-np.max([nbsw[:, 0], loca_bsw[:, 0]], axis=0), np.zeros(self.meshLocal.nC)], axis=0) *
+                       np.max([np.min([ntne[:, 1], loca_tne[:, 1]], axis=0)-np.max([nbsw[:, 1], loca_bsw[:, 1]], axis=0), np.zeros(self.meshLocal.nC)], axis=0) *
+                       np.max([np.min([ntne[:, 2], loca_tne[:, 2]], axis=0)-np.max([nbsw[:, 2], loca_bsw[:, 2]], axis=0), np.zeros(self.meshLocal.nC)], axis=0))]
+
+            V = np.c_[V]
+
+            I = Utils.mkvc(np.arange(indx.shape[0]).reshape([1, -1]).repeat(self.nCell, axis=1))
+            J = Utils.mkvc(indx.T)
+            P = sp.coo_matrix((Utils.mkvc(V), (I, J)),
+                              shape=(indx.shape[0], self.actvGlobal.sum()))
+            # self.P = Utils.sdiag(self.mesh2.vol[self.actind2])*P.tocsc()
+            self._P = P.tocsr()
+
+            self._shape = indx.shape[0], self.actvGlobal.sum()
+
+        return self._P
+
+    @property
+    def Ptransform(self):
+        """
+            Projection for the interpolation of the model (weighted average)
+        """
+        if getattr(self, '_Ptransform', None) is None:
+            sumW = Utils.mkvc(np.sum(self.P, axis=1)+1e-8)
```

Comment:
the 1e-8 is a tolerance to make sure we don't divide by zero? would you mind naming it as a variable for clarity? eg
```
tol = 1e-8
sumW = Utils.mkvc(np.sum(self.P, axis=1)) + tol
```